homeassistant/components/automation/__init__.py

```diff
@@ -313,16 +318,37 @@ def as_dict(self) -> Dict[str, Any]:
         return result
 
 
+class LimitedSizeDict(OrderedDict):
+    """OrderedDict limited in size."""
+
+    def __init__(self, *args, **kwds):
+        """Initialize OrderedDict limited in size."""
+        self.size_limit = kwds.pop("size_limit", None)
+        OrderedDict.__init__(self, *args, **kwds)
+        self._check_size_limit()
```

Comment:
Why do this during init ?

Reply:
Because key/value pairs may be passed to the constructor (although that is not currently used).